File: Walker App/KAppMakerExtended-main/iosApp/iosApp/Analytics/FirebaseAnalyticsImpl.swift
//
//  FirebaseAnalyticsImpl.swift
//  iosApp
//
//  Created by Mirzamehdi on 06/03/2025.
//

import Foundation
import ComposeApp
import FirebaseAnalytics
import FirebaseCore

class FirebaseAnalyticsImpl: ComposeApp.Analytics {
    
    func logEvent(event: String, params: [String : Any]?) {
        var eventParams: [String: Any] = [:]
        params?.forEach { key, value in eventParams[key] = "\(value)" }
        Analytics.logEvent(event, parameters: eventParams)
    }
    
    func setEnabled(enabled: Bool) {
        Analytics.setAnalyticsCollectionEnabled(enabled)
    }
    
}

File: Walker App/KAppMakerExtended-main/composeApp/src/androidMain/kotlin/com/measify/kappmaker/util/analytics/FirebaseAnalyticsImpl.kt
package com.measify.kappmaker.util.analytics

import android.os.Bundle
import com.google.firebase.analytics.FirebaseAnalytics

class FirebaseAnalyticsImpl(private val firebaseAnalytics: FirebaseAnalytics) : Analytics {

    override fun logEvent(event: String, params: Map<String, Any>?) {
        val bundle = Bundle().apply {
            for (entry in params ?: emptyMap()) putString(entry.key, entry.value.toString())
        }
        firebaseAnalytics.logEvent(event, bundle)
    }

    override fun setEnabled(enabled: Boolean) {
        firebaseAnalytics.setAnalyticsCollectionEnabled(enabled)
    }
}